src/transform/schema.ts

```diff
@@ -45,6 +45,19 @@ export function transformSchemaObjMap(obj: Record<string, any>, options: Transfo
   return output.replace(/\n+$/, "\n"); // replace repeat line endings with only one
 }
 
+/** make sure all required fields exist **/
+export function addRequiredProps(properties: Record<string, any>, required: string[]): string[] {
```

Comment:
This is a minor thing, but I’d recommend using `Set<string>` rather than `string[]`, so that duplicates will automatically be removed for us. (e.g. `required: ["foo", "foo", "foo", "bar"]`). I don’t think it’s invalid to repeat properties in `required`, but the generated TypeScript will be incorrect if this happens (I know that this is on the schema author, but the error will be really weird so I’d rather just avoid it).